rali/rali/source/meta_node_ssd_random_crop.cpp

```diff
@@ -0,0 +1,237 @@
+    vxCopyArrayRange((vx_array)_x2, 0, _batch_size, sizeof(uint), _x2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_y2, 0, _batch_size, sizeof(uint), _y2_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+
+    x_drift_factor = _meta_crop_param->get_x_drift_factor();
+    y_drift_factor = _meta_crop_param->get_y_drift_factor();
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 6);
+    const std::vector<std::pair<float, float>> IOU = {std::make_pair(0.0f, 1.0f), std::make_pair(0.1f, 1.0f), std::make_pair(0.3f, 1.0f),
+                                            std::make_pair(0.5f, 1.0f), std::make_pair(0.45f, 1.0f), std::make_pair(0.35f, 1.0f), std::make_pair(0.0f, 1.0f) };   
+    int sample_option;
+    std::pair<float, float> iou;
+    float min_iou, max_iou;
+    in_width = _meta_crop_param->in_width;
+    in_height = _meta_crop_param->in_height;
+    bool invalid_bboxes = true;
+    _enitire_iou = true;
+    BoundingBoxCord crop_box, jth_box;
+    jth_box.x = jth_box.y = jth_box.w = jth_box.h = 0; // Initializing to supress warnings. 
+    for (int i = 0; i < _batch_size; i++)
+    {
+        int bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        std::vector<int> labels_buf(bb_count);
+        memcpy(labels_buf.data(), input_meta_data->get_bb_labels_batch()[i].data(), sizeof(int) * bb_count);
+        std::vector<float> coords_buf(bb_count * 4);
+        memcpy(coords_buf.data(), input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        crop_box.h = _crop_height_val[i];
+        crop_box.w = _crop_width_val[i];
+        crop_box.x = _x1_val[i];
+        crop_box.y = _y1_val[i];
+        while (true)
+        {
+            sample_option = dis(gen);
+            iou =  IOU[sample_option];
+            min_iou = iou.first;
+            max_iou = iou.second;
+            if (!sample_option)
+            {
+                crop_box.x = 0;
+                crop_box.y = 0;
+                crop_box.h = in_height[i];
+                crop_box.w = in_width[i];
+                break;
+            }
+
+            for (int j = 0; j < _num_of_attempts; j++)
+            {
+                x_drift_factor->renew();
+                float factor = 0.3f;
+                auto w_factor = factor + (x_drift_factor->get() * (1 - factor));
+                crop_box.w = w_factor * in_width[i];
+                y_drift_factor->renew();
+                y_drift_factor->renew();
+                auto h_factor = factor + (y_drift_factor->get() * (1 - factor));
+                crop_box.h = h_factor * in_height[i];
+                //aspect ratio check
+                if ((crop_box.w / crop_box.h < 0.5) || (crop_box.w / crop_box.h > 2.))
+                    continue;
+                break;
```

Comment:
If no crop is obtained which satisfies the aspect_ratio criterion, need to go back to beginning of while loop for next choice